lib/tasks.js

```diff
@@ -111,37 +111,75 @@ module.exports = function restfulMap(req,res){
         return arySQL;
     };
 
-    this.applyTransaction = function(obj,callback){
+    this.applyTransaction = function(obj, callback, errCallback){
+
         if(!callback) callback = function(response,obj){
             res.send(response);
         };
 
         try {
+
             var arySQL = self.extractSQL(obj);
 
             //console.log(arySQL);
             console.log('Complete write');
+
             $sql.connect().then(function (connection) {
                 connection.transaction(arySQL).then(function () {
+
                     connection.release();
                     response = {
                         "status": "success"
                     };
                     callback(response,obj);
+
                 }, function (err) {
+
                     console.log(err);
                     connection.release();
+
+                    if (errCallback) {
+
+                        return errCallback(err);
+
+                    }
+
                     self.errorHandler.error(400, err.code, (err.details || err.message) + '');
+
                     return self.errorHandler.response();
+
                 });
 
             }, function (err) {
+
                 console.log(err);
+
+                if (errCallback) {
+
+                    return errCallback(err);
+
+                }
+
                 res.send(err);
+
             });
-        }catch(e){
+
+        } catch (e) {
+
             console.log(e);
+
+            if (errCallback) {
+
+                return errCallback(e);
+
+            }
+
+            self.errorHandler.error(500, e + '');
+
+            return self.errorHandler.response();
```

Comment:
will this modify current behavior? I feel we might be introducing a breaking change with this addition.

Reply:
@onel0p3z this is a bug in my understanding, any errors making it to this catch block should have been reported or responded to. I can't think of a scenario where the past way of handling is desirable?

Reply:
>  there might be logic/behavior that's already dependant on this

@onel0p3z opposite can also be true that this might be resulting in errors which might be going unreported/ not bubbled up so far. But I am happy to split it into a separate PR, so that rest of changes can be merged.

Reply:
Also @onel0p3z if you check the same code on restfulize_promise that's exactly how it has been handled there as well.